src/realm/object-store/sync/impl/sync_metadata.cpp

```diff
@@ -252,12 +238,105 @@ SyncMetadataManager::SyncMetadataManager(std::string path, bool should_encrypt,
         object_schema->persisted_properties[2].column_key, object_schema->persisted_properties[3].column_key,
         object_schema->persisted_properties[4].column_key,
     };
+}
 
-    object_schema = realm->schema().find(c_sync_app_metadata);
-    m_app_metadata_schema = {
-        object_schema->persisted_properties[0].column_key, object_schema->persisted_properties[1].column_key,
-        object_schema->persisted_properties[2].column_key, object_schema->persisted_properties[3].column_key,
-        object_schema->persisted_properties[4].column_key};
+void SyncMetadataManager::perform_launch_actions(SyncFileManager& file_manager) const
+{
+    auto realm = get_realm();
+
+    // Perform our "on next startup" actions such as deleting Realm files
+    // which we couldn't delete immediately due to them being in use
+    auto actions_table = ObjectStore::table_for_object_type(realm->read_group(), c_sync_fileActionMetadata);
+    for (auto file_action : *actions_table) {
+        SyncFileActionMetadata md(m_file_action_schema, realm, file_action);
+        run_file_action(file_manager, md);
+    }
+
+    // Delete any users marked for death.
+    auto users_table = ObjectStore::table_for_object_type(realm->read_group(), c_sync_userMetadata);
+    for (auto user : *users_table) {
+        if (user.get<int64_t>(m_user_schema.state_col) != int64_t(SyncUser::State::Removed))
+            continue;
+        try {
+            SyncUserMetadata data(m_user_schema, realm, user);
+            file_manager.remove_user_realms(data.identity(), data.realm_file_paths());
+            realm->begin_transaction();
```

Comment:
should we delete all users in a single transaction? The downside would be that in case of an exception (other than FileAccessError), no user is removed.